x-pack/test/observability_ai_assistant_functional/common/intercept_request.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { WebDriver } from 'selenium-webdriver';
+
+interface ResponseFactory {
+  fail: (reason?: string) => ['Fetch.failRequest', { requestId: string }];
+}
+
+export async function interceptRequest(
```

Comment:
this uses the chrome devtools protocol to intercept and fail a network request. i need this for the knowledge base.